From d5d4f60c835e5e92e7b766e946c330d1c006847d Mon Sep 17 00:00:00 2001
From: Michael Wright <michaelwr@google.com>
Date: Fri, 26 Jan 2018 00:16:41 +0000
Subject: [PATCH 078/306] Use correct brightness when manual brightness is
 enabled.

We want to use whatever the current brightness setting value is, not
whatever the user set last. This means that we'll have a reasonable
value at first boot and we won't get weird brightness jumps when users
disable autobrightness.

Fixes: 72508527
Test: Manual

Change-Id: I4178593f91b231ebcffb6e2104c515e557d975fc
(cherry picked from commit 7bb223428d979ba7882c7e177af2933faaffda45)
---
 .../com/android/server/display/DisplayPowerController.java   | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index b82e7c68798..c5bf6d85685 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -839,11 +839,8 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
         }
 
         // Apply manual brightness.
-        // Use the current brightness setting from the request, which is expected
-        // provide a nominal default value for the case where auto-brightness
-        // is not ready yet.
         if (brightness < 0) {
-            brightness = clampScreenBrightness(mLastUserSetScreenBrightness);
+            brightness = clampScreenBrightness(mCurrentScreenBrightnessSetting);
         }
 
 
-- 
2.17.1

